### DISCUSSION TITLE
Detect clicking on a line number

### DISCUSSION THREAD

**Discussion by ArturoDent** [microsoft/vscode-discussions#1173] [Extension Development QnA] [answered]
Is it possible to know when a line number is clicked on?

**Comment by alefragnani** [comment, marked as answer]
Hi @ArturoDent ,

If you mean the line number displayed in the gutter, then no, there is no `click` event of any sort. If you mean to know when the user clicks anywhere in the text, there is no event either, but you can use `vscode.window.onDidChangeTextEditorSelection` event, looking at the `event` property to be `TextEditorSelectionChangeKind.Mouse`.

On the other hand, if you want to know when the line number in the gutter is clicked, to add something to the context menu, similar to the `Add Breakpoint` and `Copy vscode.dev link`. entries, you must contribute to the `"editor/lineNumber/context"` entry in the `"menus"` contribution point.

Hope this helps

**Reply by ArturoDent** [reply]
Thanks, I will see if the `TextEditorSelectionChangeKind.Mouse` will work for me.  Maybe together with determining that the entire line was selected - as clicking on the line number will do.
